MidiBtlePairing/src/main/java/com/mobileer/example/midibtlepairing/DeviceScanActivity.java

```diff
@@ -286,11 +311,11 @@ public View getView(int i, View view, ViewGroup viewGroup) {
     }
 
     // Device scan callback.
-    private BluetoothAdapter.LeScanCallback mLeScanCallback =
-            new BluetoothAdapter.LeScanCallback() {
-
+    private ScanCallback mScanCallback = new ScanCallback() {
         @Override
-        public void onLeScan(final BluetoothDevice device, int rssi, byte[] scanRecord) {
+        public void onScanResult(int callbackType, ScanResult result) {
+            super.onScanResult(callbackType, result);
+            final BluetoothDevice device = result.getDevice();
```

Comment:
Should this code behave differently depending on "callbackType"?
For example [CALLBACK_TYPE_MATCH_LOST](https://developer.android.com/reference/android/bluetooth/le/ScanSettings#CALLBACK_TYPE_MATCH_LOST)

Maybe also look at result.isConnectable()
https://developer.android.com/reference/android/bluetooth/le/ScanResult#isConnectable()

Maybe a good test would be to run the app on phone A and list the LE devices.
Then list devices using the app on phone B.

Then connect phone B to the LE device A and see if phone B removes the LE device from its list.

Reply:
Created issue #87. Submitting this to address the issue first